operator/internal/manifests/distributor.go

```diff
@@ -134,6 +134,9 @@ func NewDistributorDeployment(opts Options) *appsv1.Deployment {
 
 	if opts.Stack.Replication != nil {
 		podSpec.TopologySpreadConstraints = append(podSpec.TopologySpreadConstraints, topologySpreadConstraints(*opts.Stack.Replication, LabelDistributorComponent, opts.Name)...)
+		if len(opts.Stack.Replication.Zones) > 0 {
+			podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, getZoneEnvVar())
+		}
```

Comment:
Is appending the `INSTANCE_AVAILABILITY_ZONE` env var safe on updates? I guess on every update of the pod spec, we will get the same variable appended each time. This will lead to excessive audit logs on duplicate environment variables. Do you recall adding this for the proxy variables?

https://github.com/grafana/loki/blob/f43dd58459239057cf8ca0dbe53b5e7ef89c7ae9/operator/internal/manifests/proxy_env.go#L50-L57

Reply:
Sorry I wanted to check what was done in proxy env before I respond. I had indeed forgotten we had added the reset Env function. It makes sense to use the same here as well. I will add that here as well.